### PR TITLE
Initial day/night cycle

### PR DESCRIPTION
**Pull request by wcarss**

### CODE REVIEW

**Comment by murrelljenna** [comment]
Hi, ports/adapters and also CI/CD?

**Comment by murrelljenna** [comment]
Could you use a pure effect system please

File: Assets/Sun.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    [SerializeField]
+    public AnimationCurve lightIntensity;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        GetComponent<Rigidbody>().angularVelocity = new Vector3(rotationSpeed, 0, 0);
+    }
+
+    // Update is called once per frame
```
Comment (murrelljenna) [inline Assets/Sun.cs:22]: Um, what purpose do these comments serve?

File: Assets/Sun.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    void Update()
+    {
+        Light sunLight = GetComponent<Light>();
```
Comment (murrelljenna) [inline Assets/Sun.cs:25]: Is a Light a monoidal datatype? If so we coudl instantiate a sunLight with a neutral value then foldLeft[MonoidalLight] to produce a Monoid[Light]

File: Assets/Scripts/LightingManager.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        RenderSettings.fogColor = Preset.FogColor.Evaluate(timePercent);
+
+        if (Sun != null)
+        {
+            Sun.color = Preset.SunColor.Evaluate(timePercent);
+            Sun.transform.localRotation = Quaternion.Euler(new Vector3((timePercent * 360f) - 90f, 170f, 0));
+            Sun.intensity = Preset.sunIntensityCurve.Evaluate(timePercent);
+        }
+
+        if (Moon != null)
```
Comment (murrelljenna) [inline Assets/Scripts/LightingManager.cs:47]: Hm, we have the Sun in product scope but if we start doing moon things we're going to need to consult with the new Moon product team

**Comment by murrelljenna** [comment]
Doing my due diligence

**Comment by murrelljenna** [comment]
Question: should we maybe split off the Moon behaviours into a separate MoonService responsible purely for answering Moon calls?

**Review by murrelljenna — APPROVED** [review]
Tentative approval pending an integration with Shopify

**Comment by wcarss** [comment]
agreed offline after discussion with moon PM that the shopify integration can wait for Q4 after our tech debt sprint

**Comment by wcarss** [comment]
other technical issues will be addressed in follow-up PR

**Pull request merged**